lib/main_common.pm

```diff
@@ -121,6 +122,12 @@ sub init_main {
     set_defaults_for_username_and_password();
     setup_env();
     check_env();
+    # We need to check image only for qemu backend, for svirt we validate image
+    # after it is copied to the hypervisor host.
+    if (check_var('BACKEND', 'qemu') && data_integrity_is_applicable()) {
+        my $errors = verify_checksum();
+        set_var('CHECKSUM_FAILED', $errors) if $errors;
```

Comment:
~~Why not record info instead?~~ Would you mind to also diag $errors?

Reply:
We do that later to diag all error at once. This is a hack for qemu to display it on the job page.

Reply:
@rwx788 right! Skipped that line :)